### PR TITLE
Add absolute imports, import/order eslint rule, and pre-commit hook

### PR DESCRIPTION
**Pull request by sliptype**
### Description

This PR:
* Adds the `app` alias to allow for absolute import paths
* Adds the `import/order` rule to organize imports
* Adds pre-commit hook for typecheck and linting
* Upgrades TypeScript to 4.4.4
* Renames module to `audius-mobile-client`

### Dragons

The pre-commit hook is using husky version 7 which is installed using an npm `prepare` script. Other people have reported issues with this in CI, so something to keep an eye on. See https://blog.typicode.com/husky-git-hooks-autoinstall/

### How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,35 +1,36 @@
...
-import { WebRefContextProvider } from './components/web/WebRef'
-import BottomBar from './components/bottom-bar'
-import MobileUploadDrawer from './components/mobile-upload-drawer'
-import EnablePushNotificationsDrawer from './components/enable-push-notifications-drawer'
-import CollectibleDetailsDrawer from './components/collectible-details-drawer'
+import { Provider } from 'react-redux'
+
+import Audio from 'app/components/audio/Audio'
+import GoogleCast from 'app/components/audio/GoogleCast'
+// import BottomBar from 'app/components/bottom-bar'
```
Comment (raymondjacobson) [inline src/App.tsx:11]: commented out

File: src/components/signon/SignOn.tsx [hunk 2]
```diff
@@ -367,10 +366,10 @@ const SignOn = ({ navigation }: SignOnProps) => {
       setPassword('')
 
       if (isSignin) {
-        remindUserToTurnOnNotifications(setPushNotificationsReminderVisible)
+        remindUserToTurnOnNotifications(dispatch)
```
Comment (raymondjacobson) [inline src/components/signon/SignOn.tsx:369]: is this ok?

**Review by raymondjacobson — APPROVED** [review]

File: src/App.tsx (hunk 1 above)
Comment (sliptype) [reply src/App.tsx:11]: It's still supposed to be commented out bc we don't want to show the RN BottomBar yet

File: src/components/signon/SignOn.tsx (hunk 2 above)
Comment (sliptype) [reply src/components/signon/SignOn.tsx:369]: Yup this change is tested, just a minor refactor to keep things dry

**Pull request merged**